sip-security/src/test/java/de/ikor/sip/foundation/security/authentication/SIPAuthenticationProviderTest.java

```diff
@@ -23,15 +26,15 @@
   @Mock private SIPTokenValidator<SIPBasicAuthAuthenticationToken> validator;
 
   @Test
-  void WHEN_ctor_WITH_nullTokenType_THEN_exception() throws Exception {
+  void WHEN_ctor_WITH_nullTokenType_THEN_NPE() throws Exception {
```

Comment:
Ouch!
NPE should be avoided by verification and throwing some more suitable exception. 
The source code should be revisited instead of testing for NPE

Reply:
actually not, a nullpointerexception is totally fine here, if you give null as an argument for something that is not nullable a nullpointer exception should be thrown. The same thing would happen if you use lombok @nonnull annotations

Reply:
I've added lombok config which makes @NonNull throw IllegalArgumentException instead of NPE

Reply:
https://stackoverflow.com/a/8160

Reply:
also keep in mind, that Preconditions (guava) and also Verify (apache commons) also throw nullpointexceptions in these cases

Reply:
Hi Thomas,
Thanks for your feedback. At first I was not fully aware where and why we throw NPE.
Anyway, the the comment you provided and the whole stackoverflow discussion is very interesting. We discussed on team level about the suitable code style in this case, and still decided to go with IAE. 
If you are convinced that we are making big mistake, please let's continue discussion on some other channel.